File: dodam-system-domain/dodam-domain-rds/src/main/java/b1nd/dodam/domain/rds/nightstudy/repository/NightStudyProjectRepository.java
package b1nd.dodam.domain.rds.nightstudy.repository;

import b1nd.dodam.domain.rds.nightstudy.entity.NightStudyProject;
import b1nd.dodam.domain.rds.support.enumeration.ApprovalStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDate;
import java.util.List;

public interface NightStudyProjectRepository extends JpaRepository<NightStudyProject, Long> {

    List<NightStudyProject> findByStartAtLessThanEqualAndEndAtGreaterThanEqual(LocalDate today1, LocalDate today2);

    List<NightStudyProject> findByStatusNotAndStartAtLessThanEqualAndEndAtGreaterThanEqualOrderByRoom(ApprovalStatus status, LocalDate today1, LocalDate today2);

}

File: services/service-night-study/src/main/kotlin/com/b1nd/dodamdodam/nightstudy/domain/project/repository/NightStudyProjectRepository.kt
package com.b1nd.dodamdodam.nightstudy.domain.project.repository

import com.b1nd.dodamdodam.nightstudy.domain.nightstudy.enumeration.NightStudyStatus
import com.b1nd.dodamdodam.nightstudy.domain.project.entity.NightStudyProjectEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDate
import java.util.UUID

interface NightStudyProjectRepository : JpaRepository<NightStudyProjectEntity, Long> {

    fun findAllByUserId(userId: UUID): List<NightStudyProjectEntity>

    fun findAllByStatus(status: NightStudyStatus): List<NightStudyProjectEntity>

    fun findAllByStatusAndStartAtLessThanEqualAndEndAtGreaterThanEqual(
        status: NightStudyStatus,
        startAt: LocalDate,
        endAt: LocalDate
    ): List<NightStudyProjectEntity>
}
